addons/website_forum/controllers/website_forum.py

```diff
@@ -88,41 +88,53 @@ def _get_forum_port_search_options(self, forum=None, tag=None, filters=None, my=
             'tag': str(tag.id) if tag else None,
             'filters': filters,
             'my': my,
+            'author_id': author_id,
+            'include_answers': include_answers,
         }
 
-    @http.route(['/forum/<model("forum.forum"):forum>',
+    @http.route(['/forum/all',
+                 '/forum/all/page/<int:page>',
+                 '/forum/<model("forum.forum"):forum>',
                  '/forum/<model("forum.forum"):forum>/page/<int:page>',
                  '''/forum/<model("forum.forum"):forum>/tag/<model("forum.tag"):tag>/questions''',
                  '''/forum/<model("forum.forum"):forum>/tag/<model("forum.tag"):tag>/questions/page/<int:page>''',
                  ], type='http', auth="public", website=True, sitemap=sitemap_forum)
-    def questions(self, forum, tag=None, page=1, filters='all', my=None, sorting=None, search='', **post):
+    def questions(self, forum=None, tag=None, page=1, filters='all', my=None, sorting=None, search='', author_id=False, include_answers=False, **post):
         Post = request.env['forum.post']
 
+        author = request.env['res.users'].browse(int(author_id)).exists()
+
+        if author == request.env.user:
+            my = 'mine'
         if sorting:
             # check that sorting is valid
-            # retro-compatibily for V8 and google links
+            # retro-compatibility for V8 and google links
             try:
                 sorting = werkzeug.urls.url_unquote_plus(sorting)
                 Post._generate_order_by(sorting, None)
             except (UserError, ValueError):
                 sorting = False
 
         if not sorting:
-            sorting = forum.default_order
+            sorting = forum.default_order if forum else 'bump_date desc'
```

Comment:
How does that work if we click on "Activity" to have the opposite order? And does this mean we could also not change the current `_order` on the model?

Reply:
Sorting on "Activity" is defined in the template. Here it is for when there is no sorting provided (route pointing to any forum but with no search query). It falls back to forum.default_order when a forum is provided. When there is no forum '/forum/all', we need to specifiy the sorting. We choose `bump_date desc` which will be the default sorting for any forum posts list ([#PR2592](https://github.com/odoo-dev/odoo/pull/2592))